### PR TITLE
Add chkfile support for pysisyphus

### PR DESCRIPTION
**Pull request by henryw7**
Connect the chkfile related functions from pyscf into gpu4pyscf.

### CODE REVIEW

File: gpu4pyscf/scf/hf.py [hunk 1]
```diff
@@ -419,6 +418,11 @@ def _kernel(mf, conv_tol=1e-10, conv_tol_grad=None,
         _, mf_diis.Corth = mf.eig(fock, s1e)
     else:
         mf_diis = None
+
+    if dump_chk and mf.chkfile:
```
Comment (sunqm) [inline gpu4pyscf/scf/hf.py:422]: `dump_chk` can be disabled by default. dumping chkfile can noticeably slow down the scf function.

Comment (henryw7) [reply gpu4pyscf/scf/hf.py:422]: I think it's a better idea to set `self.chkfile = None` at __init__ function of SCF class (which requires a copy of __init__ function from pyscf.hf.SCF to gpu4pyscf.hf.SCF). If we change the function signature of this internal kernel function, it'll be very hard to turn in on from pysisyphus side.

Comment (sunqm) [reply gpu4pyscf/scf/hf.py:422]: All work. This flag just needs to be muted by default.

File: gpu4pyscf/scf/tests/test_rhf.py [hunk 2]
```diff
@@ -264,11 +265,23 @@ def test_rhf_d4(self):
...
+        mf = scf.RHF(mol)
+        mf.chkfile = ftmp.name
+        mf.kernel()
+        dm_stored = mf.make_rdm1(mf.mo_coeff, mf.mo_occ)
+        dm_stored = cupy.asnumpy(dm_stored)
+
+        mf_copy = scf.RHF(mol)
+        mf_copy.chkfile = ftmp.name
+        dm_loaded = mf_copy.init_guess_by_chkfile()
+        assert np.allclose(dm_stored, dm_loaded, atol = 1e-14)
```
Comment (wxj6000) [inline gpu4pyscf/scf/tests/test_rhf.py:279]: 1e-14 is probably too strict. 1e-8 or 1e-9 should be enough for this test.

File: gpu4pyscf/scf/tests/test_uhf.py [hunk 3]
```diff
@@ -272,13 +273,28 @@ def test_uhf_d4(self):
...
+        mf = scf.UHF(mol)
+        mf.chkfile = ftmp.name
+        mf.kernel()
+        dma_stored, dmb_stored = mf.make_rdm1(mf.mo_coeff, mf.mo_occ)
+        dma_stored, dmb_stored = cupy.asnumpy(dma_stored), cupy.asnumpy(dmb_stored)
+
+        mf_copy = scf.UHF(mol)
+        mf_copy.chkfile = ftmp.name
+        dma_loaded, dmb_loaded = mf_copy.init_guess_by_chkfile()
+        assert np.allclose(dma_stored, dma_loaded, atol = 1e-14)
```
Comment (wxj6000) [inline gpu4pyscf/scf/tests/test_uhf.py:288]: tol for density matrix can be 1e-5 or 1e-6

File: gpu4pyscf/scf/tests/test_rhf.py (hunk 2 above)
Comment (henryw7) [reply gpu4pyscf/scf/tests/test_rhf.py]: I added a comment about it. The two density matrices are not from two SCF calculations, but from the MO coefficients before storage and after storage and reload. This threshold suppose to be zero, I just allow some numerical noise.

**Review by wxj6000 — APPROVED** [review]

**Pull request merged**